lib/src/data/datasource/remote/movie_api_service.dart

```diff
@@ -0,0 +1,104 @@
+import 'dart:async';
+import 'dart:convert';
+
+import 'package:http/http.dart' show Client;
+
+import '../../../core/util/constants.dart' as constants;
+import '../../../core/util/enums.dart';
+import '../../../domain/entity/genre.dart';
+import '../../../domain/entity/movie.dart';
+import '../../model/genre_model.dart';
+import '../../model/tmdb_item_model.dart';
+
+class MovieApiService {
+  String _getEndpointUrl(MovieEndpoint endpoint) {
+    switch (endpoint) {
+      case MovieEndpoint.topRated:
+        return constants.endpointTopRated;
+      case MovieEndpoint.upcoming:
+        return constants.endpointUpcoming;
+      case MovieEndpoint.nowPlaying:
+        return constants.endpointNowPlaying;
+      case MovieEndpoint.popular:
+        return constants.endpointPopular;
+    }
+  }
```

Comment:
Here instead to separate the endpoints with a enum, you can create a enum class with parameters like
```
enum MovieEndpoint {
     topRated(MovieEndpoint.endpointTopRated),
     upcoming(MovieEndpoint.endpointUpcoming),
     nowPlaying(MovieEndpoint.endpointNowPlaying),
     popular(MovieEndpoint.endpointPopular);
     
     const MovieEndpoint(String url);
     
     final String url;
}
```

Reply:
great tip, thanks!